src/unix/uilocale.cpp

```diff
@@ -295,7 +330,20 @@ wxUILocaleImplUnix::Use()
 wxString
 wxUILocaleImplUnix::GetName() const
 {
-    return m_locId.GetName();
+    wxString name = m_locId.GetName();
+    if (name.empty())
+    {
+        char* rv = setlocale(LC_ALL, NULL);
+        if (rv)
+            name = rv;
```

Comment:
I wonder what is the encoding of this string. Should we use `FromUTF8()` here?

Reply:
As far as I can tell `setlocale` always returns pure ASCII strings. Using `FromUTF8` should do no harm, so I will add it.